### DISCUSSION TITLE
Best Settings With Shunshine?

### DISCUSSION THREAD

**Discussion by pillepalle123** [VirtualDrivers/Virtual-Display-Driver#59] [Q&A] [answered]
First of all: freaking amazing work @itsmikethetech ! I have been waiting for a solution like this for years.

I'd like to know the best moonshine settings to use the virtual display driver with. I have a WQHD display connected to my PC that does not support HDR nor the 4k res that my Moonlight device (Nvidia shield with TV) has. I see two options:

1st: Have a second virtual screen constantly enabled that matches the client resolution and HDR settings. In Moonlight change the to-be-streamed monitor to the virtual one (adapter name and output name e.g.: \\.\DISPLAY4). The downside is that the PC now constantly renders a second high-resolution monitor that is not used when not streaming. Also: how do I make the Apps open on the virtual display and not the main one?

2nd: Use DisplaySwitch in Moonlight. Have a second virtual screen that matches the client resolution and HDR settings but is disabled and add the DO command "displayswitch.exe /external" and UNDO command "displayswitch.exe /internal" within each application. Downside, if something goes wrong with the streaming or I can't end the session on my client I will be locked on a black screen on my PC.

Are those two options and their settings correct or is there another way?

Thanks a lot!

**Comment by GamingBoiYT** [comment, marked as answer]
Use https://github.com/Nonary/MonitorSwapAutomation and https://github.com/Nonary/ResolutionAutomation/releases . 
These add a command to sunshine that disables the virtual monitor when not streaming, then when you start a moonlight stream, it disables your real monitor and enables the virtual one. For help on how to set these up, everything is listed on their github pages.
